core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerAggregator.java

```diff
@@ -101,18 +102,16 @@ public void aggregate(String groupKey, Iterator<StructuredRecord> iterator,
     builder.set(conf.getKeyField(), groupKey);
     while (iterator.hasNext()) {
       StructuredRecord record = iterator.next();
-      String outputFieldName = record.get(fieldName);
-      String outputFieldValue = record.get(fieldValue);
+      String outputFieldName = record.get(nameField);
+      String outputFieldValue = record.get(valueField);
 
-      if (outputFieldName == null) {
-        if (!record.getSchema().getField(fieldName).getSchema().isNullable()) {
-          throw new IllegalArgumentException("null value found for non-nullable field " + fieldName);
-        }
+      if (outputFieldName == null && record.getSchema().getField(nameField) != null && !record.getSchema().getField
```

Comment:
when outputFieldName is null, you need to skip this record. The way it is now, it will result in a null pointer when you call `builder.set(outputFieldName, outputFieldValue)`